PrebidMobile/PrebidMobileCore/src/main/java/org/prebid/mobile/core/CacheManager.java

```diff
@@ -140,4 +117,81 @@ private String getCacheForSDK(String cacheId) {
         }
         return null;
     }
+
+    private static class BidCleanUpRunnable implements Runnable {
+        private final WeakReference<CacheManager> cacheManagerWeakRef;
+        private final Handler handler;
+
```

Comment:
and only objects with references to `Context` objects are the ones we make `WeakReference`s of so they can be cleared if their original `Context` instances are no longer in use, this happens quite a lot when `Activities` are destroyed (app minimized, phone rotated, user went home, user locked screen)